### PR TITLE
Implement TVTB algebra

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**

Transposed Vector-derived Transformation Binding (TVTB) uses the same binding method as VTB, but transposes the derived matrix. This makes the identity and inverse two-sided.

**Interactions with other PRs:**

Based on #265 and PR is against that branch to get the relevant diff.

**How has this been tested?**

Mostly existing generic algebra tests, but tests for the VTB network have been copied and used to test the TVTB network.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Where should a reviewer start?**

I'd suggest to take a look at `TvtbAlgebra` first (compare it to `VtbAlgebra`) and then look at the `TVTB` network (and compare it to the `VTB` network).

**Types of changes:**

- New feature (non-breaking change which adds functionality)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [x] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

### CODE REVIEW

File: nengo_spa/algebras/tests/test_algebras.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_binding_and_invert(algebra, d, sidedness, rng):
                 warnings.simplefilter("error", DeprecationWarning)
                 if binding_side is ElementSidedness.LEFT:
                     bound = algebra.bind(b, a)
-                    r = algebra.bind(bound, algebra.invert(b, sidedness=sidedness))
+                    r = algebra.bind(algebra.invert(b, sidedness=sidedness), bound)
```
Comment (Seanny123) [inline nengo_spa/algebras/tests/test_algebras.py:55]: I'm guessing this was an error that wasn't evident until TVTB and it's sidedness was implemented?

File: nengo_spa/algebras/tvtb_algebra.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        sub_d = np.sqrt(d)
+        return sub_d * sub_d == d
+
+    def _get_sub_d(self, d):
+        sub_d = int(np.sqrt(d))
+        if sub_d * sub_d != d:
+            raise ValueError("Vector dimensionality must be a square number.")
+        return sub_d
+
+    def make_unitary(self, v):
```
Comment (Seanny123) [inline nengo_spa/algebras/tvtb_algebra.py:84]: Is there a name or explanation for this algorithm?

File: nengo_spa/algebras/tests/test_algebras.py (hunk 1 above)
Comment (jgosmann) [reply nengo_spa/algebras/tests/test_algebras.py:55]: Yes, the left-inverse needs to be applied from the left.

File: nengo_spa/algebras/tvtb_algebra.py (hunk 2 above)
Comment (jgosmann) [reply nengo_spa/algebras/tvtb_algebra.py:84]: Maybe ... probably? It's generating an orthogonal matrix, same algorithm as the original VTB.

Comment (jgosmann) [reply nengo_spa/algebras/tvtb_algebra.py:84]: Is it the [Gram-Schmidt process](https://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process)?

Comment (Seanny123) [reply nengo_spa/algebras/tvtb_algebra.py:84]: I think I even left the same comment when reviewing the original VTB...

Once again, upon further consideration, I think I can trust the name `make_unitary` even if I don't completely grasp the internals.

**Review by Seanny123 — APPROVED** [review]

**Pull request merged**